#167484603 Fetch all available profiles

malaba6 opened the pull request:
### What does this PR do?
- Creates the endpoint to fetch all the user profiles
- Tests the endpoints
- Adds the user's endpoint in the API documentation
- Fixes the `user can successfully login test`

### Description of Task to be completed?
- Create a method called `fetchProfiles` inside the class `Profile` in the `profileController` file.
- Add the route in the `routes` folder to fetch profiles with the `get` method
- In the `userProfile.test` file, add tests for this endpoint
- In `swagger.json` file, add `/api/v1/profiles` in the swagger file's path to document this endpoint

### How should this be manually tested?
In Postman or Swagger or any other tool of your choice visit this enpoint:
`/api/v1/profiles` to fetch a list of profiles

### Any background context you want to provide:
To access this endpoint, User must be authenticated. And use the token generated for Loggin or Registration as `Authorisation` in the 'header'.

### What are the relevant pivotal tracker stories?
[List user functionality](https://www.pivotaltracker.com/story/show/167484603)

### Screenshots? 
*In Postman*
![Screen Shot 2019-08-14 at 04 22 29](https://user-images.githubusercontent.com/50101879/63011626-fbbfee00-be88-11e9-8c25-7df656766edf.png)

*In swagger*
<img width="1440" alt="Screen Shot 2019-08-14 at 04 20 36" src="https://user-images.githubusercontent.com/50101879/63011657-0a0e0a00-be89-11e9-8e09-8320c8a3d28f.png">

raymond42 committed:
[Start #167484590] add user profile

raymond42 committed:
[Fixes #167484590] implementing feedback

raymond42 committed:
[Fixes #167484590] implementing feedback

Commit:
feat(fetch profiles): fetch all users profiles
This allows an authenticated user to fetch other users profiles
[Finishes #167484603]

Commit:
feat(fetch profiles): fetch all users profiles
This allows an authenticated user to fetch other users profiles
[Finishes #167484603]

harerakalex commented on src/controllers/profileController.js:
```diff
@@ -0,0 +1,126 @@
+       * @param {object} res
+       * @returns {object} return object containing updated user profile
+       */
+  static async editProfile(req, res) {
```
According to the title of this PR, `#167484603 Fetch all available profiles`, and then this method is to edit profile. Why did you create it?

malaba6 replied:
I did not create that method. I just edited the class where that method was created, Raymond created it

raymond42 reviewed (changes requested):
Working good. But first, there are some conflicts needed to be resolved

malaba6 commented:
> Working good. But first, there are some conflicts needed to be resolved

On it. This happened because of the merging. Let me rebase and get rid of the conflict

Commit:
feat(fetch profiles endpoint): endpoint to fetch all profiles
This allows an authenticated user to fetch all available profiles
[Delivers #167484603]

malaba6 committed:
Merge branch 'development' into ft-fetch-all-profiles-#167484603

Commit:
fix(fix swagger): fix a bug in swagger

Commit:
fix(fix swagger): fix a bug in swagger

mkiterian commented on src/controllers/profileController.js:
```diff
@@ -42,11 +42,32 @@ class Profile {
+        data: users
+      });
+    } catch (err) {
+      // console.log(err);
```
do not mute the error. I'd rather you throw the error, this is important for debugging purposes

malaba6 replied:
Okay...fixed!

mkiterian commented on src/tests/userprofile.test.js:
```diff
@@ -40,6 +42,68 @@ describe('GET /api/v1/profiles', () => {
+});
+
+describe('GET /api/v1/profiles', () => {
+  it('Should A list of profiles if  user is identified', (done) => {
```
grammar: "Should return a list ..."

malaba6 replied:
Alright! done!

mkiterian commented on src/tests/userprofile.test.js:
```diff
@@ -40,6 +42,68 @@ describe('GET /api/v1/profiles', () => {
+  });
+});
+
+describe('GET /api/v1/profiles', () => {
```
this looks like a copy of https://github.com/andela/coding-geeks-ah-backend/pull/19/files#diff-7e80fd915c1e3097f4adf33a8a3c7f0dR60

mkiterian replied:
I'd also like to see an assertion about the number of profiles returned is what we expect

malaba6 replied:
> this looks like a copy of https://github.com/andela/coding-geeks-ah-backend/pull/19/files#diff-7e80fd915c1e3097f4adf33a8a3c7f0dR60

Ok...fixed

malaba6 replied:
> I'd also like to see an assertion about the number of profiles returned is what we expect

On it...

mkiterian reviewed (changes requested):
I also think the `/api/profiles` endpoint shouldn't require authentication. You can also remove the auth requirement on `/api/profiles/:username`

malaba6 commented:
> I also think the `/api/profiles` endpoint shouldn't require authentication. You can also remove the auth requirement on `/api/profiles/:username`

I authenticated this endpoint because it is required according to the story description on the PT board here 
<img width="364" alt="Screen Shot 2019-08-15 at 19 57 11" src="https://user-images.githubusercontent.com/50101879/63115501-97d81b00-bf97-11e9-891c-2eb0ea63219f.png">

for `/api/profiles/:username`, I removed the authentication before I pushed, so it does not require authentication

Commit:
fix(clean code): clean code to throw arrors when caught
